clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -250,6 +251,29 @@ void AllocaOp::build(::mlir::OpBuilder &odsBuilder,
   odsState.addTypes(addr);
 }
 
+LogicalResult AllocaOp::verify() {
+  auto modOp = getOperation()->getParentOfType<ModuleOp>();
+  // If no data layout info specified, just skip it
+  if (!mlir::dyn_cast_or_null<mlir::DataLayoutSpecAttr>(
+          modOp->getAttr(mlir::DLTIDialect::kDataLayoutAttrName))) {
+    return success();
+  }
+
+  // TODO: do not initialize every time
+  ::cir::CIRDataLayout dataLayout{modOp};
```

Comment:
It's not initialized every time, it probably uses value semantics like the rest of MLIR

Reply:
https://github.com/llvm/clangir/blob/1c0064b98bbb54a0a2ec2c55785fc595d640eb1b/clang/lib/CIR/CodeGen/CIRGenModule.h#L135-L139

I used to read these lines and wanted to track it likewise. But if you confirm that it's the right way to use this class, I'm willing to remove the TODO ; )

Reply:
you can drop that comment